pennylane/templates/layers/random.py

```diff
@@ -221,22 +161,84 @@ def circuit_rnd(weights):
 
         >>> np.allclose(first_call, second_call)
         >>> True
+
+        **Parameter shape**
+
+        The expected shape for the weight tensor can be computed with the static method
+        :meth:`~.RandomLayers.shape` and used when creating randomly
+        initialised weight tensors:
+
+        .. code-block:: python
+
+            shape = RandomLayers.shape(n_layers=2, n_rotations=3)
+            weights = np.random.random(size=shape)
     """
-    if seed is not None:
-        np.random.seed(seed)
-
-    if rotations is None:
-        rotations = [RX, RY, RZ]
-
-    wires = Wires(wires)
-    repeat = _preprocess(weights)
-
-    for l in range(repeat):
-        random_layer(
-            weights=weights[l],
-            wires=wires,
-            ratio_imprim=ratio_imprim,
-            imprimitive=imprimitive,
-            rotations=rotations,
-            seed=seed,
-        )
+
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "A"
+
+    def __init__(
+        self,
+        weights,
+        wires,
+        ratio_imprim=0.3,
+        imprimitive=None,
+        rotations=None,
+        seed=42,
+        do_queue=True,
+    ):
+
+        self.seed = seed
+        self.rotations = rotations or [qml.RX, qml.RY, qml.RZ]
+
+        shape = qml.math.shape(weights)
+        if len(shape) != 2:
+            raise ValueError(f"Weights tensor must be 2-dimensional; got shape {shape}")
+
+        self.n_layers = shape[0]
+        self.imprimitive = imprimitive or qml.CNOT
+        self.ratio_imprimitive = ratio_imprim
+
+        super().__init__(weights, wires=wires, do_queue=do_queue)
+
+    def expand(self):
+
+        if self.seed is not None:
+            np.random.seed(self.seed)
+
+        shape = qml.math.shape(self.parameters[0])
+
+        with qml.tape.QuantumTape() as tape:
+
+            for l in range(self.n_layers):
+
+                i = 0
+                while i < shape[1]:
+                    if np.random.random() > self.ratio_imprimitive:
+                        # apply a random rotation gate to a random wire
+                        gate = np.random.choice(self.rotations)
+                        rnd_wire = self.wires.select_random(1)
+                        gate(self.parameters[0][l, i], wires=rnd_wire)
+                        i += 1
+
+                    else:
+                        # apply the entangler to two random wires
+                        if len(self.wires) > 1:
+                            rnd_wires = self.wires.select_random(2)
+                            self.imprimitive(wires=rnd_wires)
+        return tape
```

Comment:
Do we care at all that expanding the tape will give different circuits each time?

Reply:
That's the idea of this template...to avoid, the user must feed a seed.